user-service/src/main/java/com/movie/recommender/user/config/JwtUtil.java

```diff
@@ -0,0 +1,40 @@
+package com.movie.recommender.user.config;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+public class JwtUtil {
+
+    @Value("${jwt.secret}")
+    private String secretKey;
+
+    @Value("${jwt.expiration}")
+    private Long expirationTime;
+
+    public String generateToken(String username) {
+        return Jwts.builder()
+                .setSubject(username)
+                .setIssuedAt(new Date())
+                .setExpiration(new Date(System.currentTimeMillis() + expirationTime))
+                .signWith(SignatureAlgorithm.HS256, secretKey)
+                .compact();
+    }
+
+    public String extractUsername(String token) {
+        return Jwts.parser().setSigningKey(secretKey).parseClaimsJws(token).getBody().getSubject();
```

Comment:
extract `Jwts.parser().setSigningKey(secretKey).parseClaimsJws(token)` to a separate method, such as `getClaims(String token)` (for example), to improve readability and reuse across methods
and add exception handling to this method with logging for any JWT parsing errors